### ISSUE TITLE
Invalid type on json schema causes crash

### ISSUE THREAD

**Issue by kadu-vido** [anelendata/tap-rest-api#2]
On manually editing the json schema for the example, I changed all variables of type "number" to "sting" (rather than "string"). This causes the tap to fail with message:

> `UnboundLocalError: local variable 'filtered' referenced before assignment`

This is because on [json2schema.py](https://github.com/anelendata/tap-rest-api/blob/881a667a672ffe1d861f9a68f1658b3f819b0744/tap_rest_api/json2schema.py) types are handled between lines 172 and 213, with no failsafe for invalid types. 

I think the most helpful way to handle this would be adding something like the following before line 213:
```python
else:
  raise Exception("Schema file X contains invalid type Y")
```

Let me know if you want me to go ahead and make that PR.

**Comment by daigotanaka** [comment]
@kadu-vido sure, that would be great

**Comment by kadu-vido** [comment]
@daigotanaka getting `ERROR: Permission to anelendata/tap-rest-api.git denied to kadu-vido.`

**Comment by daigotanaka** [comment]
@kadu-vido Can you fork (not clone), make the change, push your repo to your remote, then create a pull request? That is the usual way of inter account/organization collaboration on GitHub. https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork

**Comment by kadu-vido** [comment]
Ah, my bad, hadn't done that across orgs before. Will do

**Comment by kadu-vido** [comment]
@daigotanaka singer project is being shelved at my org, so I won't be working on it any longer. 

For what it's worth, merely including the following exception at line 213 solved my issue locally:
```python
        else:
            raise Exception("Invalid type in schema: %s" % obj_type)
```

**Comment by daigotanaka** [comment]
@kadu-vido thx for letting me know. this is gong to be fixed with #3
